File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        l,r,n = 0,0,len(chars)
        while r<n:
            curr = chars[r]
            count = 0
            while r<n and curr==chars[r]:
                r+=1
                count+=1
            chars[l] = curr
            l+=1
            if count>1:
                digits = str(count)
                chars[l:l+len(digits)] = digits
                l+= len(digits)
        return l

File: 443-string-compression/string-compression.ts
function compress(chars: string[]): number {
    let l = 0
    let r = 0
    while (r<chars.length){
        let curr = chars[r]
        let count = 0
        while ( r<chars.length && curr == chars[r]){
            r++
            count++
        }
        chars[l] = curr
        l++
        if (count >1){
            let digits = count.toString()
            for (let i = 0;i<digits.length;i++){
                chars[l] = digits[i]
                l++ 
            }
        }
    }
    return l;
};